### PR TITLE
Improve the Selection Panel with better title, context, and Space View key properties

### LINKED ISSUE #4159: Rework the Selection Panel's top content
The top section of the Selection Panel has multiple issues:
- The selected thing is often highlighted in blue (because... well... it's selected), which is confusing (looks interactive when its actually not).
- The selected thing sometimes appears second instead of first.
- The enclosing context sometimes appears in the Blueprint section instead of the top section.
- The styling is inconsistant with the left sidebar.

It should be reworked as follows:
- [ ] The selected thing should be displayed as a (full width) title bar using `ListItem`, with all possible context (icon if Space View, note on the nature of the thing, etc.—see figjam).
- [ ] Make the enclosing context consistent by moving it to the top, just after the title bar.

Relevant FigJam: https://www.figma.com/file/4ypyTDAAu1RmFcPocpiPB5/Selection-Panel-cleanup?type=whiteboard&node-id=0-1&t=eKgC0PjaOa3woiE2-0

### LINKED ISSUE #4160: Add an explicit "Name" field to Selection Panel when object can be renamed (space view)
This is currently done as part of the Selection Panel's top section, which will be reworked by #4159. There should be an explicit field in the Blueprint section.

### LINKED ISSUE #4288: Display the parent item in a consistent way in the Selection Panel
The way the parent item(s) is displayed in the Selection Panel can be improved:
- The parent sometime appears _before_ the actual selected item
- The enclosing context sometimes appears in the Blueprint section instead of the top section.

The top section should be reworked such that the parent item is constantly display, right after the title.

FigJam: https://www.figma.com/file/4ypyTDAAu1RmFcPocpiPB5/Selection-Panel-cleanup?type=whiteboard&node-id=0-1&t=eKgC0PjaOa3woiE2-0

### PR DESCRIPTION
**Pull request by abey79**
### What

~~Do not merge until after 0.11.~~

This PR reworks the Selection Panel to improve the title ("What is selected?"), the context ("Where is the selected thing to be found?"), and the key properties of Space Views (name, origin, type). This is a first step towards the WIP designs by @martenbjork related to blueprints.

Changes:
- Full-width, blue title bar consistent with other panel's style and with the idea that this thing is selected.
- "Context" section listing the place(s) this thing is to be found.
  - This is much more consistant across `Item` kinds. Previously, stuff was displayed all over the place.
  - From StreamEntity, it's now it's possible to navigate to any related SpaceViewEntity _or_ Space Views. Previously, a button confusingly named after the space view would actually lead to the SpaceViewEntity.
- For Space Views, the Name/Origin/Type are now displayed on top (for all types of space views).
  - This greatly increase the discoverability of the "Name" property. Previously it was hiding in plain sight in what appeared to be contextual information.  

**Note**: the end-game for context is bread-crumbs like these, but that's for another day as we're don't yet  have an easy way to walk _up_ the hierarchy:

image

<br/>
<br/>

* Closes #4159
* Closes #4288
* Closes #4160

### Screenshots

Space View:
image

Group:

image

Entity in Space View:

image

Entity in Streams:

image

Component:

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/4324) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4324)
- [Docs preview](https://rerun.io/preview/ed064713de2471df1cac9eeeea2173a4f4fd7c7e/docs) 
- [Examples preview](https://rerun.io/preview/ed064713de2471df1cac9eeeea2173a4f4fd7c7e/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by nikolausWest** [comment]
From a first test, I think it's a pretty straight forward improvement over what we have now. I guess the only thing that is a slight negative is that we no longer write out what the type of the thing you've selected is. Since this info exists on hover and we intend to indicate the type also through a new set of icons in the future I don't think that should be a blocker. It

Good stuff!

**Comment by abey79** [comment]
@nikolausWest I don't disagree. As a matter of fact, my "design" included explicit type:

image

We shortly discussed that with @martenbjork who preferred to not overload this title bar. We can certainly revisit this and iterate.

**Comment by emilk** [comment]
> Do not merge until after 0.11.

why not?

**Comment by abey79** [comment]
> why not?

I was seeing this as a first step towards a bigger story, but yeah if we like it we can roll it in 0.11—no technical reason not to. We probably shouldn't _delay_ 0.11 for this though.

**Comment by abey79** [comment]
@emilk This is not new, but got more visible with the blue background. I had hoped the floating scroll bars would magically fix it, but it's apparently not the case. I will look into it—at minima, open an issue if we dont have one.

**Review by emilk — APPROVED** [review]
I think the `// NOLINT` is a hint that we're doing something wrong, but otherwise LGTM

File: crates/re_viewer/src/ui/selection_panel.rs [hunk 1]
```diff
@@ -100,26 +102,34 @@ impl SelectionPanel {
...
                 }
 
-                ctx.re_ui
-                    .large_collapsing_header(ui, "Blueprint", true, |ui| {
-                        blueprint_ui(ui, ctx, viewport, item);
-                    });
-
-                if i + 1 < num_selections {
-                    // Add space some space between selections
-                    ui.add(egui::Separator::default().spacing(24.0).grow(20.0));
```
[the remark this answers is not in the document]
Comment (abey79) [reply crates/re_viewer/src/ui/selection_panel.rs:122]: I added 8. spacing

File: crates/re_viewer/src/ui/selection_panel.rs [hunk 2]
```diff
@@ -166,72 +178,199 @@ fn what_is_selected_ui(
...
-                        .on_hover_text(component_name.full_name());
-                    ui.end_row();
-                });
+            item_title_ui(
+                ctx.re_ui,
+                ui,
+                component_name.short_name(),
+                None,
+                &format!(
+                    "Component {} of entity \"{}\"", // NOLINT
```
[the remark this answers is not in the document]
Comment (abey79) [reply crates/re_viewer/src/ui/selection_panel.rs]: I switched to single quotes for the time being.

**Pull request merged**